### PR TITLE
redshift: postgres: use schema names and delimited identifiers

### LINKED ISSUE #224: Postgres only supports public schema
The queries for Postgres tables limits to only the public schema. 

- https://github.com/plotly/falcon-sql-client/blob/master/backend/persistent/datastores/Sql.js#L36
- https://github.com/plotly/falcon-sql-client/blob/master/backend/persistent/datastores/Sql.js#L169

I'm open to create a PR, but my attempts to run the application locally have been unsuccessful because of OAuth issues when connecting to the plotly cloud.

### LINKED ISSUE #246: Tables within schema not visible in side-panel when using Redshift as a data source
Tables within a schema other than public are not visible in Query side-panel when using Redshift as a data source. However, tables within schemas other than public are queryable if the user knows the name of the table(s) beforehand.

### LINKED ISSUE #252: redshift: surround table name with quote marks
See #250 and http://docs.aws.amazon.com/redshift/latest/dg/r_names.html

### LINKED ISSUE #254: Don't skip Redshift tests in routes.spec.js

### PR DESCRIPTION
**Pull request by n-riesco**
@tarzzz  would you review this one?

* same as postgres (see #250).
* restored redshift tests in `routes.spec.js`
* added redshift case in the schemas test.

Closes #254 
Closes #252 
Closes #246 
Closes #224

### CODE REVIEW

File: backend/persistent/datastores/Sql.js [hunk 1]
```diff
@@ -39,13 +39,13 @@ const SHOW_TABLES_QUERY = {
...
-        'table_schema != \'pg_catalog\' and ' +
-        'table_schema != \'information_schema\''
-    )
+    [DIALECTS.REDSHIFT]: `
+        SELECT table_name
+        FROM information_schema.tables
+        WHERE table_type != 'VIEW'
+           AND table_schema != 'pg_catalog'
+           AND table_schema != 'information_schema'
+    `
```
Comment (tarzzz) [inline backend/persistent/datastores/Sql.js]: 👍

File: test/backend/routes.spec.js [hunk 2]
```diff
@@ -695,129 +742,566 @@ describe('Routes:', () => {
...
-                            [ 'walmart_store_openings_1962_2006', 'date_super', 'character varying' ],
-                            [ 'walmart_store_openings_1962_2006', '1', 'integer' ],
-                            [ 'weather_data_seattle_2016', 'min_temperaturec', 'character varying' ],
-                            [ 'weather_data_seattle_2016', 'mean_temperaturec', 'character varying' ],
-                            [ 'weather_data_seattle_2016', 'max_temperaturec', 'character varying' ],
-                            [ 'weather_data_seattle_2016', 'date', 'character varying' ],
-                            [ 'world_gdp_with_codes_2014', 'code', 'character varying' ],
-                            [ 'world_gdp_with_codes_2014', 'gdp (billions)', 'numeric' ],
-                            [ 'world_gdp_with_codes_2014', 'country', 'character varying' ]
+                            [ 'my_new_topo."edge"', 'edge_id', 'integer' ],
```
Comment (tarzzz) [inline test/backend/routes.spec.js]: Entering the large data in tests here is not good.. Can you please open an issue to move these large, static datasets to a separate file.. `spec_constant.js` or something?

**Comment by n-riesco** [comment]
@tarzzz Since there are already other specs using large datasets, I'd rather do it in a different PR.

I've opened issue #258 for that.

**Comment by n-riesco** [comment]
@tarzzz I'm an idiot. I've been waiting for the :dancer: and today I realised you already approved the review. I wish github would send an email when a PR gets approved.

**Pull request merged**